doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-Linux/index.md

```diff
@@ -2,304 +2,186 @@
-cd build
-cmake -DKAA_DEBUG_ENABLED=1 -DKAA_MAX_LOG_LEVEL=4 -DKAA_WITHOUT_EVENTS=1 ..
-make
-make install
-~~~
+This section is about how to build your Kaa aplication using C SDK.
 
-# C++ endpoint SDK
+In this guide we use *CMake* as a build system for the application, so focus on the writing the build system rather than the application's code. If you want master writing applciation code go to the Programming guide section. Before going on, make shure that all [dependencies](#dependencies) are installed.
 
-## Install build utils and dependencies
+1. First of all let's create some directories. The directories determine build structure. To keep it clean and transparent create three directories: my-kaa-application - a root directory, which contains all files, my-kaa-application/kaa - for the C SDK source files, my-kaa-application/src - for the application code.
 
-Before building the C++ endpoint SDK, install the following components on your machine:
+        mkdir my-kaa-application
+        cd my-kaa-application
+        mkdir kaa
+        mkdir src
 
-1. Install compilers:
-    1. For automatic installation, execute the following commands:
+2. Generate C SDK and unpack it to my-kaa-application/kaa directory:
 
-       ~~~ shell
-       $ sudo apt-get install g++
-       ~~~
+        tar -xvf c-sdk-archive.tar.gz -C my-kaa-application/kaa
 
-    2. For manual installation of version 4.8, refer to the following example:
+3. Create the application code. Create kaa-application.c file in my-kaa-application/src directory:
 
-       ~~~ shell
-       $ sudo apt-get install python-software-properties
-       $ sudo add-apt-repository ppa:ubuntu-toolchain-r/test
-       $ sudo apt-get update
-       $ sudo apt-get install g++-4.8
-       $ sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-4.8 50
-       ~~~
+        touch src/kaa-application.c
 
-2. Install CMake utility:
-    1. For automatic installation, execute the following commands (tested on Ubuntu 14.04):
+4. Open kaa-application.c and write the application code:
 
-       ~~~ shell
-       $ sudo apt-get install cmake
-       ~~~
 
-    2. For manual installation, refer to the following example:
+        #include <stdio.h>
+        #include <stdlib.h>
+        #include <kaa/kaa.h>
+        #include <kaa/platform/kaa_client.h>
+        #include <kaa/kaa_error.h>
 
-       ~~~ shell
-       $ wget http://www.cmake.org/files/v3.3/cmake-3.3.0-rc2.tar.gz
-       $ tar -zxf cmake-3.3.0-rc2.tar.gz
-       $ cd cmake-3.3.0-rc2/
-       $ ./configure
-       $ sudo make install
-       ~~~
 
-3. Install the [**Boost** ](http://www.boost.org/users/download/)libraries:
-    1. For automatic installation, execute the following commands:
+        #define KAA_APP_PRINT_TIMES 5
 
-       ~~~ shell
-       $ sudo apt-get install libboost1.55-all-dev
-       ~~~
+        static void dummy_function(void *context)
+        {
+            for (size_t i = 1; i < KAA_APP_PRINT_TIMES; i++) {
```

Comment:
Are you sure wee really need to print that more than once? After all, this is an introduction to Kaa SDK, and not C language :)